### PR TITLE
added BundleAnalyzerPlugin & separated vendor bundle

### PR DESCRIPTION
**Pull request by kuldeepkeshwar**

### CODE REVIEW

**Comment by kuldeepkeshwar** [comment]
@gdi2290 can you help me out with the commits here ?
my changes are 
	merged upstream			ed08cef
 	added lazy route			9f88d69
 	vcode			a4b6899

but it is showing old changes also

**Comment by Hongbo-Miao** [comment]
@kuldeepkeshwar can u fork the latest commit and create a PR based on it?

**Comment by kuldeepkeshwar** [comment]
updated the PR

File: src/app/lazy/lazy-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { NgModule } from '@angular/core';
+import { RouterModule ,Routes} from '@angular/router';
+
+import { LazyComponent } from './lazy.component';
+const routes:Routes=[];
```
Comment (gdi2290) [inline src/app/lazy/lazy-routing.module.ts]: will this work for AoT?

Comment (kuldeepkeshwar) [inline src/app/lazy/lazy-routing.module.ts]: yes it will

**Comment by maufarinelli** [comment]
Just my 2 cents: I have tried this changes in my fork, but my bindings are not working and I didn't figure out why. I put an input with [(ngModel)] and a (click) binding in elements and they are not working anymore.

**Comment by kuldeepkeshwar** [comment]
@maufarinelli can you share the link to your repo ? I have tested it , it is working

**Comment by John0x** [comment]
I have the same problem as @maufarinelli. 
@kuldeepkeshwar I tried your repo (https://github.com/kuldeepkeshwar/universal-starter). The bindings don't seem to work. Only tested the home component.
Added a button and an event, but it never gets called. It never prints "Works!" (see code below).

````
import { Component } from '@angular/core';

import { ModelService } from '../shared/model/model.service';

@Component({
  selector: 'home',
  template: `
  Home component
  <button (click)="testExp()">Test</button>
  {{ data | json }}
  `
})
export class HomeComponent {
  data = {};
  constructor(public model: ModelService) {

    // we need the data synchronously for the client to set the server response
    // we create another method so we have more control for testing
    this.universalInit();
  }

  universalInit() {
    this.model.get('/data.json').subscribe(data => {
      this.data = data;
    });
  }

  testExp() {
    console.log('Works!');
  }

}
````

**Comment by maufarinelli** [comment]
@kuldeepkeshwar the repo I am testing it is: 
https://github.com/maufarinelli/universal-starter-multientries/tree/lazy-load 
Lazy Load branch

**Comment by kuldeepkeshwar** [comment]
updated the pr , removed lazy routes and added BundleAnalyzerPlugin & separated vendor bundle.

**Comment by John0x** [comment]
@kuldeepkeshwar do lazy routes not work with universal currently? Saw them working in qdoubles starter: https://github.com/qdouble/angular-webpack2-starter

**Comment by kuldeepkeshwar** [comment]
Lazy route works with non AoT version only

**Comment by kuldeepkeshwar** [comment]
If you want , I can share example for non AoT

**Comment by John0x** [comment]
@kuldeepkeshwar right now or is it intended to not work by the universal team?

**Comment by MarkPieszak** [comment]
We want everything to work :8ball: @John0x 
Just at the moment, certain things need to be fixed in Core, or Universal itself. :/

**Comment by gdi2290** [comment]
@kuldeepkeshwar can you make a separate PR for BundleAnalyzerPlugin? I won't add vendor until later when webpack adds async tag support. Sorka is working async tag support as the default for webpack 2

**Comment by kuldeepkeshwar** [comment]
Cool , will make another pr

**Comment by gdi2290** [comment]
thanks

**Pull request closed without merging**